### PR TITLE
Automatically sync notifications after signup

### PR DESCRIPTION
**Pull request by andrew**
Fixes https://github.com/andrew/octobox/issues/49

### CODE REVIEW

**Comment by andrew** [comment]
Yeah, just trying to remember how to fire the after_commit callbacks in tests 🙈

**Comment by andrew** [comment]
@tarebyte any tips on how to test this?

**Comment by tarebyte** [comment]
I would just create an object and make an assertion that the API endpoint was called.

**Comment by andrew** [comment]
I've pushed a failing test, it's midnight here and my brain is broken!

File: test/models/user_test.rb [hunk 1]
```diff
@@ -63,4 +63,18 @@ class UserTest < ActiveSupport::TestCase
       assert_equal n.archived, true
     end
   end
+
+  test "triggers sync_notifications on save" do
+    notifications_url = %r{https://api.github.com/notifications}
+
+    body     = '[]'
+    headers  = { 'Content-Type' => 'application/json' }
+    response = { status: 200, body: body, headers: headers }
```
Comment (tarebyte) [inline test/models/user_test.rb]: This is starting to become a pattern, we may want to look into a helper for this.

**Comment by tarebyte** [comment]
@andrew so the `after_commit` ActiveRecord callback was making the call recursive so it was an infinite loop 😆 

I've updated it to use `after_create` which works and has the same functionality ¯\\_(ツ)_/¯

**Comment by andrew** [comment]
Ahhh that would explain it, thanks for the sanity check!

**Pull request merged**